packages/preview-middleware/src/base/flp.ts

```diff
@@ -145,13 +148,51 @@ export class FlpSandbox {
             local: '.',
             intent: this.config.intent
         });
-
         this.addStandardRoutes();
+        if (this.rta) {
+            this.rta.options ??= {};
+            this.rta.options.baseId = componentId ?? manifest['sap.app'].id;
+            this.addEditorRoutes(this.rta);
+        }
         this.addRoutesForAdditionalApps();
         this.logger.info(`Initialized for app ${manifest['sap.app'].id}`);
         this.logger.debug(`Configured apps: ${JSON.stringify(this.templateConfig.apps)}`);
     }
 
+    /**
+     * Add additional routes for configured editors.
+     *
+     * @param rta runtime authoring configuration
+     */
+    private addEditorRoutes(rta: RtaConfig) {
+        for (const editor of rta.editors) {
+            let previewUrl = editor.path;
+            if (editor.developerMode) {
+                previewUrl = `${previewUrl}.inner.html`;
+                editor.pluginScript ??= 'open/ux/preview/client/cpe/init';
+                this.router.get(editor.path, (async (_req: Request, res: Response) => {
+                    const template = readFileSync(join(__dirname, '../../templates/flp/editor.html'), 'utf-8');
+                    const html = render(template, {
+                        previewUrl: `${previewUrl}?sap-ui-xx-viewCache=false&fiori-tools-rta-mode=forAdaptation&sap-ui-rta-skip-flex-validation=true&sap-ui-xx-condense-changes=true#${this.config.intent.object}-${this.config.intent.action}`
+                    });
+                    res.status(200).contentType('html').send(html);
+                }) as RequestHandler);
+            }
+            this.router.get(previewUrl, (async (_req: Request, res: Response) => {
+                const config = { ...this.templateConfig };
+                config.flex = {
+                    layer: rta.layer,
+                    ...rta.options,
```

Comment:
Is passing all options from the yaml file to RTA intentional?

Reply:
This is not coming from the yaml. The parameter options can only be set programmatically and is to be used to set scenario specific settings e.g. when initializing ADP, we set the `projectId`